apps/docs/src/hooks/useCommentDraft.ts

```
@@ -0,0 +1,384 @@
+export function useCommentDraft({
+  pageUrl,
+  parentId,
+  onRecover,
+}: UseCommentDraftOptions): UseCommentDraftReturn {
+  const [draft, setDraftState] = useState("");
+  const [hasDraft, setHasDraft] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+  const [lastSaved, setLastSaved] = useState<Date | null>(null);
+  const [isEnhancing, setIsEnhancing] = useState(false);
+
+  const saveTimeoutRef = useRef<NodeJS.Timeout | null>(null);
```

ESLint

[error] 'NodeJS' is not defined.

(no-undef)